src/Rings/groebner.jl

```diff
@@ -1221,3 +1232,104 @@ function _compute_groebner_basis_using_fglm(I::MPolyIdeal,
 	dim(I) == 0 || error("Dimension of ideal must be zero.")
 	I.gb[destination_ordering] = _fglm(G, destination_ordering)
 end
+
+
+# TODO: does the target ordering have to be global?
+@doc Markdown.doc"""
+    _groebner_basis_with_hilbert(I::MPolyIdeal,
```

Comment:
Make this an exported function called `groebner_basis_hilbert_driven()`, we will later on adjust calls to `f4` and `fglm` to this naming convention. Here you correctly have the `weights` as parameters, but we do not have them for the general `..._basis()` calls.